### ISSUE TITLE
Unable to upload new revision after first revision uploaded and activated

### ISSUE THREAD

**Issue by jamesdixon** [duizendnegen/ember-cli-deploy-azure#4]
Hi,

I've installed `ember-cli-deploy` and your addon into a brand new Ember project. I was successful in deploying and activating the first revision. However, after I've made a change and then subsequently run `ember deploy` again, I see the following error:

```
Upload failed!

Did you try to upload an already uploaded revision?
```

I don't see any documentation on having to explicitly tell `ember deploy` to create a new revision, but maybe I'm missing something. Appreciate the assistance!

Best,
James

**Comment by duizendnegen** [comment]
Hi James,

Great to hear that you had success with deploying the first revision! Did the C# server-side code also work for you?
ember-cli-deploy (and also ember-deploy-azure) works under the assumption that the underlying project is a Git-backed project. It calculates the revision hash from that as well, with which it stores the index.html file. If a revision with a similar hash is uploaded (most likely the same one with minor amendments, as I experience frequently - but it could also be a hash collision) it rejects the newly uploaded file. A similar issue has also been raised in https://github.com/ember-cli/ember-cli-deploy/issues/105.

Long story short: you have to create a new git commit with your changes, so that the uploadable hash changes, then try again.
Let me know if that solves your issue!

**Comment by jamesdixon** [comment]
Doh! It worked! 

Thank you very much for the explanation. That would definitely explain it -- I was curious as to how Ember CLI was detecting changes (or not). 

As for the C# code, I'm actually running Node and had to write some code to handle the pull from Azure Tables. That said, your C# code was a great guide! I plan on publishing it and linking back to your repo for those who may also be using the Node/Azure combo.

Thanks again!

Best,
James

**Comment by duizendnegen** [comment]
Good stuff! Glad it's solved.
Let me know when you publish it, I'll link it from the main page then :)

**Comment by jamesdixon** [comment]
Hi again, here's the link to the package I published to handle pulling from Azure Tables in Node/Express: [https://github.com/jamesdixon/node-ember-cli-deploy-azure-tables](https://github.com/jamesdixon/node-ember-cli-deploy-azure-tables). Linked to your addon for those in need!

**Comment by duizendnegen** [comment]
Cool, I'm going to update the readme anyway over the weekend, will then also feature the two currently existing server-side approaches.
